### PR TITLE
Allow columns to not have transformer

### LINKED ISSUE #473: Allow columns to not have a transformer
### Problem Description

It should be possible to skip the transformation of select columns by specifying that no transformation is needed in the config.

### Expected behavior

- Use `None` in place of a transformer object to specify that no transformer is needed for the column
- Add new methods `remove_transformers` and `remove_transformers_by_sdtype` for better usability in removing transformers from the config
- When a column is marked as `None`, the column is carried over as-is during the `transform` call; no changes in value or column name

### Additional context

This change will touch many methods.

**`set_config`**: Allow the use of `None` in place of a transformer object
```python
import rdt.transformers as rt

ht = HyperTransformer()
ht.set_config(config={
  'column_A': rt.FloatFormatter(missing_value_replacement=0.00),
  'column_B': None, # don't transform this column
  'column_C': rt.FrequencyEncoder()
})
```

**`update_transformers`** and **`update_transformers_by_sdtype`**: Also allow the use of `None`, though this is not the preferred approach
```python
ht = HyperTransformer()
ht.detect_initial_config(data)

# the following methods will work but will not be mentioned in the docs
ht.update_transformers(column_name_to_transformer={
  'column_B': None
})

ht.update_transformers_by_sdtype(sdtype='categorical', transformer=None)
transformed = ht.fit_transform(data)
```

**`remove_transformers`** and **`remove_transformers_by_sdtype`**: The preferred way to remove transformers from the config
```python
ht = HyperTransformer()
ht.detect_initial_config(data)

# remove transformers for the given list of column names
ht.remove_transformers(column_names=['column_B'])

# remove the transformers for everything in the given sdtype
ht.remove_transformers_by_sdtype(sdtype='categorical')

transformed = ht.fit_transform(data)
```

**`get_config`**: Return (and print out) `None` for the appropriate transformer(s)
```python
{
  'sdtypes': {
    'column_A': 'float',
    'column_B': 'datetime',
    'column_C': 'categorical'
  },
  'transformers': {
    'column_A': FloatFormatter(missing_value_replacement=0.00)
    'column_B': None
    'column_C': FrequencyEncoder()
  }
}
```

#### 1. Invalid sdtype
Check the `sdtype` the user passes into `remove_transformers_by_sdtype`
```python
ht.remove_transformers_by_sdtype(sdtype='unkown_type')
Error: Invalid sdtype '<name>'. If you are trying to use a premium sdtype, contact info@sdv.dev about RDT Add-Ons.
```

#### 2. Invalid column name
```python
ht.remove_transformers(column_names=['column_A', 'unknown_column'])
Error: Invalid column names: ['unknown_column'...]. These columns do not exist in the config. Use 'get_config()' to see the expected values
```

#### ~~3. Sdtype & transformer are not compatible~~
_Ignore this case. It does not apply for these functions._

#### 4. User tries to use either of these functions after already calling fit
```python
ht = HyperTransformer()
ht.detect_initial_config(data)
ht.fit()

ht.remove_transformers(column_names=['column_A'])
Warning: For this change to take effect, please refit your data using 'fit' or 'fit_transform'.

ht.remove_transformers_by_sdtype(sdtype='categorical')
Warning: For this change to take effect, please refit your data using 'fit' or 'fit_transform'.
```

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolve #473

### CODE REVIEW

File: rdt/hyper_transformer.py [hunk 1]
```diff
@@ -302,17 +305,69 @@ def update_transformers(self, column_name_to_transformer):
...
+                unknown_columns.append(column_name)
+            else:
+                self.field_transformers[column_name] = None
+                self._provided_field_transformers[column_name] = None
+
+        if self._fitted:
+            warnings.warn(self._REFIT_MESSAGE)
+
+        if unknown_columns:
+            raise Error(
```
Comment (amontanez24) [inline rdt/hyper_transformer.py:342]: Should we be raising the error after the fact, or should all changes be avoided if there are any unknown columns?  @npatki Basically if some columns are valid, should those still be changed or should the method fail completely?

Comment (npatki) [reply rdt/hyper_transformer.py:336]: Yeah the entire think should error. That way the user can fix it and re-pass the entire list.

File: tests/unit/test_hyper_transformer.py [hunk 2]
```diff
@@ -679,6 +679,38 @@ def test__fit_field_transformer_multi_column_field_not_ready(
...
+    def test__fit_field_transformer_transformer_is_none(self):
+        """Test the ``_fit_field_transformer`` method.
+
+        Test that when a ``transformer`` is ``None`` the ``outputs`` are the same
+        as the field.
+
+        Setup:
+            - Dataframe with a column.
+            - HyperTransformer instance.
+
```
Comment (amontanez24) [inline tests/unit/test_hyper_transformer.py:691]: Maybe add a section for input and say that the input is `None`

File: tests/unit/test_hyper_transformer.py [hunk 3]
```diff
@@ -1983,6 +2209,31 @@ def test_get_final_output_columns(self):
         # Assert
         outputs == ['field1.out1.value', 'field1.out2.out1.value', 'field1.out2.out2.value']
 
+    def test_get_final_output_columns_transformer_is_none(self):
+        """Test the ``get_fianl_output_columns`` method.
```
Comment (amontanez24) [inline tests/unit/test_hyper_transformer.py:2213]: typo: final

**Review by amontanez24 — APPROVED** [review]
LGTM! Thanks for addressing

**Review by fealho — APPROVED** [review]

**Pull request merged**